ci/version.sh

```diff
@@ -11,8 +11,18 @@
 #     0.6.0-37-g3878937f-dirty
 #
 
-set -ex
+git_describe_output=$(git describe --tags --dirty 2>&1)
+exit_code=$?
 
-version=$(git describe --tags --dirty)
-version="${version:1}"
-echo $version > version
+set -ex 
+if [ $exit_code -ne 0 ]; then
+  echo "There were errors while getting the tags. Defaulting to 0.0.0-0"
+  version="0.0.0-0"
+  echo $version > version
+  exit 1
```

Comment:
Wouldn't this fail for you? It seems it would return a return code 1, which would exit the parent script too early?

Reply:
I am sorry. PR is still WIP. I should find a final solution this weekend. But yes, it does exit the parent script. The reason I chose this approach is because of what you said about [if the tags fail, we want to see it break](https://github.com/lcompilers/lpython/pull/2303#issuecomment-1696502164). But I can change this

Reply:
Yes. I would leave the `version.sh` script alone. But I would implement a new script that is not called by our CI, but can be used by you. Then things will fail for us (as we want it), but they would work for you (as you want it).

Reply:
Alright, `version.sh` checked out to main. Created the helper script in the root of the project. Please let me know if I should place it anywhere else. Maybe a `scripts` folder for this sort of stuff?